.github/workflows/ci.yml

```diff
@@ -1,10 +1,23 @@
 #
 # TODO: add description & license header
 #
-name: PlatformIO CI - Build Process
+name: Code Format & PlatformIO Build
 on: [push]
 jobs:
+  format:
+    name: Clang-Format
+    runs-on: ubuntu-latest
+    steps:
+      - name: install
+        run: sudo apt-get install -y clang-format
+      - name: checkout
+        uses: actions/checkout@v2
+      - name: format
+        run: find . -not -path "./.*" -name "*.cpp" -o -name "*.c" -o -name "*.h" | xargs -I {} clang-format -i {}
```

Comment:
@andreaskuster Proposel
find . -type d \( -path "./.*" -o -path "./datasheets" -o -path "./medis" \) -prune -false -o -name "*.h" -o -name ".hpp" -o -name "*.cpp" | xargs -I {} clang-format -i {}
It is the same as in [https://github.com/SolarEnergyRacers/solar_car_control_system/blob/convert_flat_c_to_cpp_classes/extras/format.sh](https://github.com/SolarEnergyRacers/solar_car_control_system/blob/convert_flat_c_to_cpp_classes/extras/format.sh)

Reply:
Sure, I am fine with that. Please commit it and merge the PR.